amr-wind/overset/TiogaInterface.cpp

```diff
@@ -105,14 +105,17 @@ void TiogaInterface::register_solution()
     m_qnode = repo.create_scratch_field(
         pres.num_comp(), pres.num_grow()[0], pres.field_location());
 
+    vel.fillpatch(0.0);
+    pres.fillpatch(0.0);
+
     field_ops::copy(*m_qcell, vel, 0, 0, vel.num_comp(), vel.num_grow());
     field_ops::copy(*m_qnode, pres, 0, 0, pres.num_comp(), pres.num_grow());
 }
 
 void TiogaInterface::update_solution()
 {
     auto& repo = m_sim.repo();
-    auto& vel = repo.get_field("velocity");
+    auto& vel = repo.get_field("velocity").state(amr_wind::FieldState::Old);
```

Comment:
Before `advance` is called, both `old` and `new` states should be same, correct? So why explicitly ask for `old`?

Reply:
because most of the calls in predictor expect the old state to be filled. So after update solution we either need to update new and advance states or just update old